### PR TITLE
ALMA: use auth in HEAD request

### PR DESCRIPTION
**Pull request by keflavich**
A HEAD request is made before the GET request, and it needs to be authorized too.

Also, there are even more URLs to try...

### CODE REVIEW

File: astroquery/query.py [hunk 1]
```diff
@@ -227,6 +227,7 @@ def _request(self, method, url, params=None, data=None, headers=None,
                 # Windows doesn't allow special characters in filenames like
                 # ":" so replace them with an underscore
                 local_filename = local_filename.replace(':', '_')
+            # should this default to savedir over cache_location?
```
Comment (keflavich) [inline astroquery/query.py:230]: I added this note as a warning to us; we might want to raise it as an Issue. I think defaulting to `cache_location` is likely the wrong choice from a user's perspective.

**Comment by bsipocz** [comment]
could you also add a changelog?

**Comment by bsipocz** [comment]
Could you rebase?

**Comment by keflavich** [comment]
I rebased, but now CI isn't running because the last commit has [CI skip].  I think we still want travis to run after the rebase, so I'm going to remove the [ci skip] comment.

**Comment by bsipocz** [comment]
Thanks @keflavich!

**Pull request merged**